src/main/utility/childpid_watcher.rs

```diff
@@ -244,6 +248,8 @@ impl ChildPidWatcher {
         let mut inner = self.inner.lock().unwrap();
         let pidfd = rustix::process::pidfd_open(pid.into(), PidfdFlags::empty())
             .unwrap_or_else(|e| panic!("pidfd_open failed for {pid:?}: {e:?}"));
+        let flags = rustix::io::fcntl_getfd(&pidfd).unwrap();
+        rustix::io::fcntl_setfd(&pidfd, flags | FdFlags::CLOEXEC).unwrap();
```

Comment:
I don't think you need to set the cloexec flag here, linux should do it automatically for `pidfd_open`:

> The  pidfd_open() system call creates a file descriptor that refers to the process whose PID is specified in pid.  The file descriptor is returned as the function result; the close-on-exec flag is set on the file descriptor.

Reply:
Good catch; replaced this with a `debug_assert` to validate this property.